dvc/repo/__init__.py

```diff
@@ -86,29 +89,16 @@ def __init__(self, root_dir=None):
 
         self.updater.check()
 
-    def __repr__(self):
-        return "Repo: '{root_dir}'".format(root_dir=self.root_dir)
-
     @staticmethod
     def find_root(root=None):
-        if root is None:
-            root = os.getcwd()
-        else:
-            root = os.path.abspath(os.path.realpath(root))
-
-        while True:
-            dvc_dir = os.path.join(root, Repo.DVC_DIR)
-            if os.path.isdir(dvc_dir):
-                return root
-            if os.path.ismount(root):
-                break
-            root = os.path.dirname(root)
-        raise NotDvcRepoError(root)
+        return Config.find_root(root=root)
```

Comment:
why is it in Config? I thought we decided to keep simple and stupid.

Reply:
why don't we use self.root_dir everywhere instead? find_root and self.roo_dir are confusing - almost the same name, hard to see the difference and where one of them should be used

Reply:
Because Config can exist outside of Repo. You mean `root_dir` as a property that would do find_root if needed and cache the result? Don't really see how `root_dir` property is almost the same as `find_root` method.

Reply:
if it's outside of repo what does root_dir mean? 

from a client perspective - which one should I use in what situations - find_root or root_dir? What is the difference between them. It's really unclear.

Reply:
config should not be implementing search for a root anyways. I see that it's probably coming from a circular dependency, but it just means that it should be solved in some other proper way. Repo calling a static method in config to search for its own root is really strange.